test/BinaryVotingPowerEarningPowerCalculator.unit.t.sol

```diff
@@ -0,0 +1,487 @@
+    _assumeSafeOracleEligibilityModule(_oracleEligibilityModule);
+
+    vm.expectRevert(
+      BinaryVotingPowerEarningPowerCalculator
+        .BinaryVotingPowerEarningPowerCalculator__InvalidVotingPowerUpdateInterval
+        .selector
+    );
+    new BinaryVotingPowerEarningPowerCalculator(
+      _owner, _oracleEligibilityModule, _votingPowerToken, 0
+    );
+  }
+}
+
+contract GetEarningPower is BinaryVotingPowerEarningPowerCalculatorTest {
+  // Oracle is available && delegate IS eligible → return sqrt(votingPower)
+  function testFuzz_AvailableOracleAndEligibleDelegateHasEarningPower(
+    address _delegate,
+    uint256 _votingPower,
+    uint256 _unusedParam1,
+    address _unusedParam2
+  ) public {
+    _setOracleAsAvailable();
+    _setDelegateeAsEligibleWithVotingPower(_delegate, _votingPower);
+
+    uint256 _expectedVotingPower = uint256(Math.sqrt(_votingPower));
+
+    assertEq(
+      calculator.getEarningPower(_unusedParam1, _delegate, _unusedParam2), _expectedVotingPower
+    );
+  }
+
+  // Oracle is paused && delegate IS eligible → return sqrt(votingPower)
+  function testFuzz_PausedOracleAndEligibleDelegateHasEarningPower(
+    address _delegate,
+    uint256 _votingPower,
+    uint256 _unusedParam1,
+    address _unusedParam2
+  ) public {
+    _setOracleAsUnavailableAsOraclePaused();
+    _setDelegateeAsEligibleWithVotingPower(_delegate, _votingPower);
+
+    uint256 _expectedVotingPower = uint256(Math.sqrt(_votingPower));
+
+    assertEq(
+      calculator.getEarningPower(_unusedParam1, _delegate, _unusedParam2), _expectedVotingPower
+    );
+  }
+
+  // Oracle is stale && delegate IS eligible → return sqrt(votingPower)
+  function testFuzz_StaleOracleAndEligibleDelegateHasEarningPower(
+    address _delegate,
+    uint256 _votingPower,
+    uint256 _unusedParam1,
+    address _unusedParam2
+  ) public {
+    _setOracleAsUnavailableAsOracleStale();
+    _setDelegateeAsEligibleWithVotingPower(_delegate, _votingPower);
+
+    uint256 _expectedVotingPower = uint256(Math.sqrt(_votingPower));
+
+    assertEq(
+      calculator.getEarningPower(_unusedParam1, _delegate, _unusedParam2), _expectedVotingPower
+    );
+  }
+
+  // Oracle is paused && delegate IS NOT eligible → return sqrt(votingPower)
+  function testFuzz_PausedOracleAndIneligibleDelegateHasEarningPower(
+    address _delegate,
+    uint256 _votingPower,
+    uint256 _unusedParam1,
+    address _unusedParam2
+  ) public {
+    _setOracleAsUnavailableAsOraclePaused();
+    _setOracleAsUnavailableAsOracleStale();
+    _setDelegateeAsNotEligibleWithVotingPower(_delegate, _votingPower);
+
+    uint256 _expectedVotingPower = uint256(Math.sqrt(_votingPower));
+
+    assertEq(
+      calculator.getEarningPower(_unusedParam1, _delegate, _unusedParam2), _expectedVotingPower
+    );
+  }
+
+  // Oracle is stale && delegate IS NOT eligible → return sqrt(votingPower)
+  function testFuzz_StaleOracleAndIneligibleDelegateHasEarningPower(
```

Comment:
Let's continue this convo https://github.com/withtally/obolstaking/pull/19#discussion_r2192418634